src/java.base/share/classes/sun/net/www/ParseUtil.java

```diff
@@ -171,8 +171,9 @@ private static byte unescape(String s, int i) {
      * Returns a new String constructed from the specified String by replacing
      * the URL escape sequences and UTF8 encoding with the characters they
      * represent.
+     * @throws IllegalArgumentException if {@code s} could not be decoded
      */
-    public static String decode(String s) {
+    public static String decode(String s) throws IllegalArgumentException {
```

Comment:
I'm not sure if you meant to add this or not but "throws IAE" is not needed, usually don't put unchecked exceptions in the throws.

Reply:
Hello Alan, that was intentional to be more prominently visible, but I'll remove it and will just retain `@throws` javadoc tag that I added as part of this change.

Reply:
Done - I've updated the PR to remove the throws clause from the method signature.